src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -431,16 +433,21 @@ std::array<bool, 2> SpatialJoinAlgorithms::isAPoleTouched(
 
 // ____________________________________________________________________________
 Result SpatialJoinAlgorithms::BoundingBoxAlgorithm() {
-  bool alreadyWarned = false;
-
-  auto printWarning = [&]() {
-    if (!alreadyWarned) {
-      AD_LOG_WARN << "expected a point here, but no point is given. Skipping "
-                     "this point and future 'non points' of this query"
-                  << std::endl;
-      alreadyWarned = true;
-    }
-  };
+  auto printWarning =
+      [alreadyWarned = false](std::optional<SpatialJoin*> spatialJoin) mutable {
```

Comment:
You can also capture the spatial join by reference (either explicitly by using `&spatialJoin = spatialJoin_` and use that in the lambda, or by capturing `this` , and using `spatialJoin_` directly in the lambda. The first one is more explicit. Then you don't need the redundant argument.